packages/mason_cli/lib/src/commands/make.dart

```diff
@@ -75,7 +75,10 @@ class _MakeCommand extends MasonCommand {
     final configPath = results['config-path'] as String?;
     final fileConflictResolution =
         (results['on-conflict'] as String).toFileConflictResolution();
+    final setExitIfChanged = results['set-exit-if-changed'] as bool;
     final target = DirectoryGeneratorTarget(Directory(outputDir));
+    final lastModified =
+        setExitIfChanged ? target.dir.statSync().modified : null;
```

Comment:
Need to think about this some more since it doesn't look like this `modified` stat isn't exactly what we want. It might make sense to adjust `generate` to return an object containing the summary of the generation including which files have been created, modified, etc...

The other option is to use https://github.com/dart-lang/watcher but it's probably not a great idea since it's not necessarily indicative of whether mason made the change or not. There could theoretically be some other process that modified the files and it would trigger a false positive.

Reply:
>  It might make sense to adjust generate to return an object containing the summary of the generation including which files have been created, modified, etc... but this would be a breaking change to `package:mason`

Would the changes in #223 be considered a breaking change (throwing an extra `MasonException` under specific conditions)? If not, maybe it's less of a change to mason?

If so, I still think that may be a less impactful change than modifying the return type of the `generate` command imho.

Reply:
Yeah will think about it some more and get back to you 👍 
On the one hand, the generate signature change would be breaking but it would still scope the --set-exit-if-changed functionality to the CLI (since it is a CLI responsibility/feature imo). The main thing I'm wondering is whether it makes sense for the core mason package to have a concept of erroring when there's a change (since the only use-case afaik would be for the CLI) 🤷

Reply:
> I'm wondering is whether it makes sense for the core mason package to have a concept of erroring when there's a change 

Ahh I see what you mean. If we adjusted the generate command to return a list of the objects that could change, and cli could see that and use it how it wanted. In this case mason cli would use it for detecting errors, but it wouldn't have to be.